python/python/bystro/proteomics/handler.py

```diff
@@ -0,0 +1,362 @@
+"""Query an annotation file and return a list of sample_ids and genes meeting the query criteria."""
+import io
+import math
+import os
+import subprocess
+from pathlib import Path
+from typing import Any, Tuple
+
+import pandas as pd
+import ray
+from bystro.proteomics.tests.example_query import DEFAULT_FIELDS  # TK
+from bystro.proteomics.utils import GNU_TAR_EXECUTABLE_NAME
+from bystro.search.utils.annotation import AnnotationOutputs, get_delimiters
+from bystro.search.utils.messages import SaveJobData
+from bystro.search.utils.opensearch import gather_opensearch_args
+from opensearchpy import OpenSearch
+from ruamel.yaml import YAML
+
+ray.init(ignore_reinit_error=True, address="auto")
+RAY_ERROR = -1  # the error code we use if a ray remote job fails to return
+
+ONE_DAY = "1d"  # default keep_alive time for opensearch point in time index
```

Comment:
We should just be extra careful to catch exceptions and clean up pit indices regardless of whether query succeeds